components/raftstore/src/store/worker/raftlog_gc.rs

```diff
@@ -137,28 +108,40 @@ impl<EK: KvEngine, ER: RaftEngine> Runner<EK, ER> {
                     "end_index" => t.end_idx,
                 );
             }
-            groups.push(RaftLogGcTask {
-                raft_group_id: t.region_id,
-                from: t.start_idx,
-                to: t.end_idx,
-            });
-        }
-        let start = Instant::now();
-        match self.gc_raft_log(groups) {
-            Err(e) => {
+            if let Err(e) = self
+                .engines
+                .raft
+                .gc(t.region_id, t.start_idx, t.end_idx, &mut batch)
+            {
                 error!("failed to gc"; "err" => %e);
-                self.report_collected(0);
                 RAFT_LOG_GC_FAILED.inc();
             }
-            Ok(n) => {
-                debug!("gc log entries";  "entry_count" => n);
-                self.report_collected(n);
-                RAFT_LOG_GC_DELETED_KEYS_HISTOGRAM.observe(n as f64);
-            }
+        }
+        fail::fail_point!("worker_gc_raft_log");
+        if let Err(e) = self.engines.raft.consume(&mut batch, false) {
+            error!("failed to write gc task"; "err" => %e);
+            RAFT_LOG_GC_FAILED.inc();
         }
         RAFT_LOG_GC_WRITE_DURATION_HISTOGRAM.observe(start.saturating_elapsed_secs());
-        for cb in cbs {
-            cb()
+        fail::fail_point!("worker_gc_raft_log_finished");
+        cbs
+    }
+
+    fn flush(&mut self) {
+        if self.tasks.is_empty() {
+            return;
+        }
+        fail::fail_point!("worker_gc_raft_log_flush");
```

Comment:
I don't see how this makes the behavior the same as past. As this is not key path, how about leave this file as it is to avoid introduce potential risk to v1?

Reply:
The interface is changed, now we don't now how many entries are compacted, so we don't report_collected anymore. Note that `report_collected` is only used in unit tests below.